server/src/com/thoughtworks/go/server/ui/PluginViewModel.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2016 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.server.ui;
+
+import java.util.List;
+
+public class PluginViewModel {
+
+    private final String id;
+    private final String name;
+    private final String version;
+    private final String type;
+    private String viewTemplate;
+    private List<PluginConfigurationViewModel> configurations;
+
+    public PluginViewModel(String id, String name, String version, String type) {
+        this.id = id;
+        this.name = name;
+        this.version = version;
+        this.type = type;
+    }
+
+    public PluginViewModel(String id, String name, String version, String type, List<PluginConfigurationViewModel> pluginConfigurationViewModels) {
+        this(id, name, version, type);
+
+        this.configurations = pluginConfigurationViewModels;
+    }
+
+    public PluginViewModel(String id, String name, String version, String type, String viewTemplate, List<PluginConfigurationViewModel> pluginConfigurationViewModels) {
+        this(id, name, version, type);
+
+        this.viewTemplate = viewTemplate;
+        this.configurations = pluginConfigurationViewModels;
+    }
```

Comment:
These sort of constructors are a huge problem, can we ensure that the constructor with less number of args, calls the constructor with more number of args? This allows for all variables to be marked final. Notice you you are not setting `configurations` but it cannot be final because of how your constructors are wired.

So for e.g. —

```

public PluginViewModel(String id, String name, String version, String type, List<PluginConfigurationViewModel> pluginConfigurationViewModels) {
  this(id, name, version, type, null, pluginConfigurationViewModels);
}

public PluginViewModel(String id, String name, String version, String type) {
  this(id, name, version, type, null, null);
}
```